postgis/lwgeom_out_mvt.c

```diff
@@ -45,26 +45,76 @@ Datum ST_AsMVTGeom(PG_FUNCTION_ARGS)
 	elog(ERROR, "Missing libprotobuf-c");
 	PG_RETURN_NULL();
 #else
-	LWGEOM *lwgeom_in, *lwgeom_out;
+	GBOX *bounds = NULL;
+	int32_t extent = 0;
+	int32_t buffer = 0;
+	bool clip_geom = true;
 	GSERIALIZED *geom_in, *geom_out;
-	GBOX *bounds;
-	int extent, buffer;
-	bool clip_geom;
+	LWGEOM *lwgeom_in, *lwgeom_out;
+	uint8_t type = 0;
+
 	if (PG_ARGISNULL(0))
+	{
 		PG_RETURN_NULL();
-	geom_in = PG_GETARG_GSERIALIZED_P_COPY(0);
-	lwgeom_in = lwgeom_from_gserialized(geom_in);
+	}
+
 	if (PG_ARGISNULL(1))
+	{
 		elog(ERROR, "%s: parameter bounds cannot be null", __func__);
-	bounds = (GBOX *) PG_GETARG_POINTER(1);
+		PG_RETURN_NULL();
+	}
+	bounds = (GBOX *)PG_GETARG_POINTER(1);
+	if (bounds->xmax - bounds->xmin <= 0 || bounds->ymax - bounds->ymin <= 0)
+	{
+		elog(ERROR, "%s: bounds width or height cannot be 0", __func__);
+		PG_RETURN_NULL();
+	}
+
 	extent = PG_ARGISNULL(2) ? 4096 : PG_GETARG_INT32(2);
+	if (extent <= 0)
+	{
+		elog(ERROR, "%s: extent cannot be 0", __func__);
```

Comment:
check is <=, message is about =0 sharp. can lead to hours of looking for 0 in code :)